Fix #21744 : [CI Failure]: Unhandled error: HttpError: Resource not accessible by integration In Full stack test

Mayank77maruti opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #21744 
2. This PR does the following: In this Workflow, We would use `pull_request_target` to solve the http error resource not found error.
3. (For bug-fixing PRs only) The original bug occurred because: `name: Leave a comment on the PR if workflow failed` this job was not able to comment on the pull request when the full_stack_test.yml failed although it was given pull request- write permission.

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [x] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [x] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [x] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).


## Proof that changes are correct

<!--
Add before-and-after videos/screenshots of the user-facing interface (including
the browser devtools console) to demonstrate that the changes made in this PR
work correctly. Make sure the actions taken in the before and after videos are
the same. (For changes involving responsiveness or adjustment of UI elements,
this should be a video that gradually resizes the viewport from wide to narrow
and back again.) If this PR is for a developer-facing feature, provide
videos/screenshots of the developer-facing interface instead.

When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->

https://github.com/WorkVed-Spaces-Actions/oppia/pull/3
As the full stack test fails it comments perfectly on the test account.

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

Mayank77maruti committed:
Update full_stack_tests.yml

Mayank77maruti commented:
@U8NWXD @seanlip PTAL
Thanks

U8NWXD commented on .github/workflows/full_stack_tests.yml:
```diff
@@ -7,7 +7,7 @@ on:
       - develop
       - release-*
-  pull_request:
+  pull_request_target:
```
This change will cause the workflow to run in the context of the PR base (generally `develop`) instead of the PR head, preventing the workflow from testing the changes in the PR ([docs](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target)).

There are ways around this limitation (e.g. checking out the PR head manually to run the tests), but the limits are in place for a good reason: a workflow that has permission (via `GITHUB_TOKEN`) to make changes to the Oppia repo shouldn't be running untrusted code. Since code in PRs hasn't been reviewed, it's inherently untrusted. I think this makes commenting on PRs in a workflow that runs tests fundamentally problematic.

U8NWXD replied:
Here are some possible solutions:

1. The workflow that runs the tests uses `pull_request` and uploads a signed artifact at the end of its run, which specifies whether the tests passed. Then, another workflow uses `pull_request_target` and waits for that artifact to appear. Once it appears, this second workflow downloads the artifact, checks whether the tests passed, and if they failed, leaves a comment on the PR. Note that to download artifacts created by other workflows, this workflow will need a PAT, not the usual GITHUB_TOKEN.
2. One workflow runs the tests using `pull_request_target`. Another workflow uses the GitHub API to query the status of CI checks on the PR and leaves a comment if any fail. This second job could even run against all open PRs at regular intervals (e.g. every minute), which might be more efficient than having a job idle for each PR with pending CI checks.

seanlip replied:
Please discuss with @Rd4dev as well -- she may have thoughts on this too, having implemented similar functionality for Oppia Android in the past.

Rd4dev replied:
Thanks for looping me in, @seanlip.

@U8NWXD, I believe this is related to posting comments on PR threads. Just adding some context since a similar issue was encountered in Oppia Android while trying to post coverage as comments. We utilized the `pull_request_target` trigger to grant read/write repository permissions (even for forks). As mentioned, since it runs in the context of the base branch, the following approaches were considered:

1. Splitting workflows into two parts (similar to the first proposed solution above):
  - Core workflow: [[code_coverage.yml](https://github.com/oppia/oppia-android/blob/08e00847d955ec1f38fe8000832a2cff1bbff62f/.github/workflows/code_coverage.yml#L3)]
  Utilizing `pull_request`
    - Executes the jobs
    - Produces a result
    - Uploads the result as an artifact
    
  - Comment workflow: [[comment_code_coverage.yml](https://github.com/oppia/oppia-android/blob/08e00847d955ec1f38fe8000832a2cff1bbff62f/.github/workflows/comment_coverage_report.yml#L3)]
  Utilizing `pull_request_target`
    - Waits for the core workflow to complete and return a conclusion (used `ArcticLampyrid/action-wait-for-workflow@v1.2.0`)
    - Identifies the last successful workflow run ID of the core workflow
    - Downloads the generated artifact using the run ID
    - Posts the comment

The intent was to keep the `pull_request_target` comment workflow as minimal as possible—handling only resource gathering and commenting—so we would rarely need to update it.

2. Testing workflows:
  - Since the permission issue primarily affects fork PRs, specific testing was required. I tested it by cloning the repository, creating a fork, merging the changes into develop, and [manually verifying the results](https://github.com/oppia/oppia-android/pull/5483#:~:text=Proof%20of%20ci%20check%20runs%20for%20every%20mentioned%20case)—though this may not be the most efficient approach.

U8NWXD reviewed (changes requested):
This problem should also affect our other tests that use the `post-comment` action, right?

Mayank77maruti commented:
So I think, the most effective approach is to implement a separate, minimally-permissioned workflow triggered by `pull_request_target` which would run when `full_stack_test.yml` is completed and `full_stack_test.yml` passes a status (success, failed) that we will query using `github api`.
This workflow will manage the task of commenting on pull requests, allowing our existing `full_stack_test.yml` workflow (triggered by `pull_request`) to remain largely untouched.
Like this: it avoids the need to alter our current testing process and maintains a clear separation between testing and commenting, and ensures that only the essential permissions are granted to the commenting workflow. 

@U8NWXD @Rd4dev Super grateful for the insights! Let me know how this looks

Mayank77maruti commented:
> So I think, the most effective approach is to implement a separate, minimally-permissioned workflow triggered by `pull_request_target` which would run when `full_stack_test.yml` is completed and `full_stack_test.yml` passes a status (success, failed) that we will query using `github api`. This workflow will manage the task of commenting on pull requests, allowing our existing `full_stack_test.yml` workflow (triggered by `pull_request`) to remain largely untouched. Like this: it avoids the need to alter our current testing process and maintains a clear separation between testing and commenting, and ensures that only the essential permissions are granted to the commenting workflow.
> 
> @U8NWXD @Rd4dev Super grateful for the insights! Let me know how this looks

Should I make progress as mentioned in here.
@U8NWXD @Rd4dev

Rd4dev commented:
> > So I think, the most effective approach is to implement a separate, minimally-permissioned workflow triggered by `pull_request_target` which would run when `full_stack_test.yml` is completed and `full_stack_test.yml` passes a status (success, failed) that we will query using `github api`. This workflow will manage the task of commenting on pull requests, allowing our existing `full_stack_test.yml` workflow (triggered by `pull_request`) to remain largely untouched. Like this: it avoids the need to alter our current testing process and maintains a clear separation between testing and commenting, and ensures that only the essential permissions are granted to the commenting workflow.
> > @U8NWXD @Rd4dev Super grateful for the insights! Let me know how this looks
> 
> Should I make progress as mentioned in here. @U8NWXD @Rd4dev

@Mayank77maruti, that's the overview of how workflows involving writing comment are laid out in Oppia-Android. Will be interested in hearing @U8NWXD's thoughts on this, as I'm not fully familiar with the specific nuances of web workflows. However, I believe it should be largely similar. 

Additionally a heads-up regarding the `ArcticLampyrid/action-wait-for-workflow` external action. In scenarios where a new contributor requires manual approval to run workflows, this action will trigger an 'action_required' conclusion causing it to fail. Once the other workflows are manually approved, the failed workflows need to be re-triggered manually. This is the tracking issue:

In ArcticLampyrid/action-wait-for-workflow: https://github.com/ArcticLampyrid/action-wait-for-workflow/issues/269

Mayank77maruti commented:
> So I think, the most effective approach is to implement a separate, minimally-permissioned workflow triggered by `pull_request_target` which would run when `full_stack_test.yml` is completed and `full_stack_test.yml` passes a status (success, failed) that we will query using `github api`. This workflow will manage the task of commenting on pull requests, allowing our existing `full_stack_test.yml` workflow (triggered by `pull_request`) to remain largely untouched. Like this: it avoids the need to alter our current testing process and maintains a clear separation between testing and commenting, and ensures that only the essential permissions are granted to the commenting workflow.
> 
> @U8NWXD @Rd4dev Super grateful for the insights! Let me know how this looks

Hey @U8NWXD Should i move forward with this approach.

seanlip commented:
Also @Mayank77maruti just a note: if you want someone to comment on your PR or answer a question, please make sure they are in the Assignees list on the right. Otherwise they might miss it. Thanks.

U8NWXD commented:
I agree with seanlip's comment approving this approach

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

Mayank77maruti committed:
Update full_stack_tests.yml

Mayank77maruti committed:
Create full_stack_test_comment.yml

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

Mayank77maruti commented:
I tested it on https://github.com/WorkVed-Spaces-Actions/oppia-test/pull/10 here and it comments perfectly.
@U8NWXD @Rd4dev @seanlip PTAL
Thanks.

Rd4dev commented on .github/workflows/full_stack_test_comment.yml:
```diff
@@ -0,0 +1,74 @@
+      - name: Leave a Comment on the PR
+        uses: peter-evans/create-or-update-comment@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Was this mandatory? I believe the action takes `GITHUB_TOKEN` as the default - https://github.com/peter-evans/create-or-update-comment?tab=readme-ov-file#action-inputs

Mayank77maruti replied:
done

Rd4dev commented on .github/workflows/full_stack_test_comment.yml:
```diff
@@ -0,0 +1,74 @@
+      pull-requests: write
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Find CI Workflow Run for PR
```
Just to confirm, is the acquired workflow `run-id` utilized anywhere? In the Oppia-Android workflow, an artifact was passed between workflows, requiring the run ID for [downloading](https://github.com/oppia/oppia-android/blob/a4317ba22be260c313b6c7d9a80abd87908d8441/.github/workflows/comment_coverage_report.yml#L102). Here, it seems like the conclusion alone was sufficient.

Mayank77maruti replied:
done

Rd4dev reviewed (commented):
Thanks @Mayank77maruti, PTAL -- just requested for few clarifications. 
Also, a reminder - as mentioned in https://github.com/oppia/oppia/pull/21831#pullrequestreview-2591429980, this will need to be implemented for all workflows that utilize `post-comment`.

Mayank77maruti committed:
Delete .github/workflows/full_stack_test_comment.yml

Mayank77maruti committed:
Create comment_on_ci_workflow.yml

Mayank77maruti committed:
Update comment_on_ci_workflow.yml

Mayank77maruti committed:
Update backend_unit_tests.yml

Mayank77maruti committed:
Update frontend_unit_tests.yml

Mayank77maruti commented:
https://github.com/WorkVed-Spaces-Actions/oppia-test/pull/15 
I tested it here and it is working perfectly.

Mayank77maruti commented:
@U8NWXD @Rd4dev PTAL
Thanks

Mayank77maruti committed:
Update backend_unit_tests.yml

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

Rd4dev commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,63 @@
+name: Comment on CI Workflows
+
```
Please include the top-level `permissions: read-all` to address the failing check. - https://github.com/oppia/oppia/actions/runs/13305628832/job/37155879933?pr=21831#step:4:723

Mayank77maruti replied:
Done

Rd4dev commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,63 @@
+              helpful.
+    steps:
+      - name: Await Test Workflow Completion
+        id: wait
```
Nit: Suggestion to provide a more descriptive `id` name for clarity.

Mayank77maruti replied:
Done

Rd4dev commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,63 @@
+          exit 1
+      - name: Notify on Test Failure
+        if: ${{ steps.wait.outputs.run-conclusion == 'failure' && !cancelled() }}
+        uses: peter-evans/create-or-update-comment@v3
```
Can you update it to the latest `v4` version?

Mayank77maruti replied:
Done

Rd4dev reviewed (commented):
Thanks @Mayank77maruti, this looks nice -- just a couple of minor suggestions, PTAL.

Mayank77maruti committed:
Update comment_on_ci_workflow.yml

Mayank77maruti commented:
@seanlip PTAL
Thanks

seanlip commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,65 @@
+      pull-requests: write
+    strategy:
+      matrix:
+        test_name: ["Full-Stack Tests", "Backend Coverage Tests", "Frontend Tests"]
```
This is meant to be a list of strings, right? If so, suggest using bullet points like the other lists:

```
test_name:
  - Full-Stack Tests
  - Backend Coverage Tests
  - Frontend Tests
```

Mayank77maruti replied:
Done

seanlip commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,65 @@
+            failure
+            action_required
+      - name: Verify Manual Approval Requirement
+        if: ${{ steps['wait_for_completion_of_workflow'].outputs.run-conclusion == 'action_required' }}
```
Does steps.wait_for_completion_of_workflow work, here and below?

Mayank77maruti replied:
Done

Rd4dev commented:
Thanks, @Mayank77maruti, for promptly addressing the comments! Moving forward, please ensure all possible demonstrations and edge cases are tested upfront and linked here, after any changes made. Since this is based on a `pull_request_target` trigger, any post-merge issues would involve redundant effort to fix.

That being said, all my comments seem addressed. @seanlip, I took an initial pass—could you help resolve them? I don't think I have the necessary permissions.

seanlip commented:
@Rd4dev -- thanks! No need to resolve the comments. On Web we don't follow that practice.

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

Mayank77maruti committed:
Update comment_on_ci_workflow.yml

Mayank77maruti commented:
@U8NWXD PTAL.
Thanks.

Mayank77maruti commented:
@U8NWXD Friendly Ping on this PR.
Thanks

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

Mayank77maruti commented:
Hey @seanlip I believe this pr is ready to be merged.
Can you please ping @U8NWXD to have a look into it once.

Mayank77maruti commented:
Hey @U8NWXD I believe this pr is ready to be merged.
Can you please take on look on this.
Thank You

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

U8NWXD commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,68 @@
+    steps:
+      - name: Await Test Workflow Completion
+        id: wait_for_completion_of_workflow
+        uses: ArcticLampyrid/action-wait-for-workflow@v1.2.0
```
Vetting for new dependency (from oppia/oppia-web-developer-docs#327):

**For Small, Un-Compiled Open-Source Dependencies:** For small open-source dependencies where the code is executed directly, we can review the source code manually.

* [x] **We have manually reviewed the source code for the dependency and confirmed that it is safe.** I manually reviewed https://github.com/ArcticLampyrid/action-wait-for-workflow/tree/7226d2d248ed9b7e231de31209d4f5f5d4c30295/src
* [x] **Dependency compilation is safe.** This is essential because if the source code of a dependency is compiled and the compiled form is downloaded and executed, we cannot inspect the compiled form to ensure it is safe. We need to confirm one of the following:
  * [ ] **The dependency is not compiled at all.** For example, shell scripts.
  * [ ] **The dependency is compiled, and we have confirmed that the inspected source code produces the pinned compiled version bit-for-bit.** For example, locally compile the inspected source code and check that the output matches what we install.
* [x] **Our reference to the dependency is pinned by a hash.** To ensure malicious code cannot be introduced after our review, we must pin a hash of the specific code we reviewed. This could be a commit hash, for example.

**Edit:** After consulting with seanlip, we've added a new set of criteria for dependencies whose source code we can audit manually. I still need to check that the dependency is compiled safely.

**Edit 2:** I've spent the last week or so trying to confirm safe dependency compilation, but I haven't been able to get the package to compile locally. @Mayank77maruti could you take over and address the two un-checked points above? Specifically, we need to:
1. Download or check out the source code I manually reviewed: https://github.com/ArcticLampyrid/action-wait-for-workflow/tree/7226d2d248ed9b7e231de31209d4f5f5d4c30295/src
2. Compile the dependency (uses ncc to transpile and combine the source code and dependencies into a single javascript file)
3. Confirm that the resulting javascript file is an exact, bit-for-bit match for https://github.com/ArcticLampyrid/action-wait-for-workflow/blob/7226d2d248ed9b7e231de31209d4f5f5d4c30295/dist/index.js. You can do this by comparing their checksums. Alternatively, if there are differences, we'll need to manually review each difference.
4. In this PR, pin our usage of this dependency to commit 7226d2d248ed9b7e231de31209d4f5f5d4c30295

Mayank77maruti replied:
**Checksum Verification**:
   - The SHA-256 checksum of the locally compiled file (`dist/index.js`) was calculated:
     ```
     4a790d46d38b86e7b49e5042b34783bace079c51f8e527b3556bf5db988bbec4
     ```
   - The checksum of the published file (`remote_index.js`) was also calculated:
     ```
     4a790d46d38b86e7b49e5042b34783bace079c51f8e527b3556bf5db988bbec4
     ```
   - Both checksums match exactly, confirming that the locally compiled file is **bit-for-bit identical** to the published file.

We are good to pin the workflow to the verified commit:

```yaml
- uses: ArcticLampyrid/action-wait-for-workflow@7226d2d248ed9b7e231de31209d4f5f5d4c30295
```

U8NWXD commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,68 @@
+          - test_name: "Full-Stack Tests"
+            workflow_file: "full_stack_tests.yml"
+            message: >
+              🛠️ Hi @${{ github.event.pull_request.user.login }}, it looks like the **Comprehensive Full-Stack Tests** are failing on your PR.
```
why are we bolding the test name here but not for the other tests below?

Mayank77maruti replied:
Removed it

U8NWXD commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,68 @@
+          exit 1
+      - name: Notify on Test Failure
+        if: ${{ steps.wait_for_completion_of_workflow.outputs.run-conclusion == 'failure' && !cancelled() }}
+        uses: peter-evans/create-or-update-comment@v4
```
Vetting for new dependency (from oppia/oppia-web-developer-docs#327):

* [x] **Actively maintained based on recent commit history:** Yes. https://github.com/peter-evans/create-or-update-comment/commit/71345be0265236311c031f5c7866368bd1eff043
* Code changes are secured by one of the following:
  * [ ] **Only a small group of people have commit access:**
  * [x] **Code review is enforced:** https://scorecard.dev/viewer/?uri=github.com/peter-evans/create-or-update-comment
* [x] **No binary artifacts:** https://scorecard.dev/viewer/?uri=github.com/peter-evans/create-or-update-comment
* [x] **Any unfixed vulnerabilities reported by GitHub are minor or not relevant:**
   * GHSA-3xgq-45jj-v275: regex DoS. Not a concern because (a) DoS would only affect the PR being evaluated and (b) the inputs are either safe (PR number) or defined by us (the message)
   * GHSA-952p-6rrq-rcjv: regex DoS
   * GHSA-c2qf-rxjj-qqgw: regex DoS
* [x] **Compatible license:** MIT https://github.com/ArcticLampyrid/action-wait-for-workflow/blob/main/LICENSE
* [x] **Adoption Scope:** No. Meets neither criterion, though used by oppia-android. https://github.com/ArcticLampyrid/action-wait-for-workflow/network/dependents
     * At least 10 dependents where each dependent has at least 100 stars on GitHub. See https://github.com/peter-evans/create-or-update-comment/network/dependents
* [x] **Established project:** Issue num 1 is from Nov 2019

LGTM. Just leaving this comment to document my reasoning for okaying this dependency

Mayank77maruti committed:
Update comment_on_ci_workflow.yml

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

Mayank77maruti commented:
Hey @U8NWXD , How should i proceed with this PR, then.
Do i need to make any changes or add anything.
Please do let me know.
Thank You

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

U8NWXD commented:
Hi @Mayank77maruti I just updated my comment https://github.com/oppia/oppia/pull/21831#discussion_r1966815603. To merge this PR we need to confirm that the `action-wait-for-workflow` dependency is safe. I manually reviewed the code and confirmed that it looks okay, but I haven't been able to compile it locally to confirm that compiling the source code actually produces the compiled dependency in `dist` that GitHub Actions will execute. Specifically, I'm getting an error from exceeding the maximum number of stack calls allowed. I also encountered a heap memory exceeded error, which I fixed by increasing the heap memory limit for javascript.

Mayank77maruti committed:
Merge branch 'develop' into http_resource_error

Mayank77maruti committed:
Update comment_on_ci_workflow.yml

Mayank77maruti commented:
> Hi @Mayank77maruti I just updated my comment [#21831 (comment)](https://github.com/oppia/oppia/pull/21831#discussion_r1966815603). To merge this PR we need to confirm that the `action-wait-for-workflow` dependency is safe. I manually reviewed the code and confirmed that it looks okay, but I haven't been able to compile it locally to confirm that compiling the source code actually produces the compiled dependency in `dist` that GitHub Actions will execute. Specifically, I'm getting an error from exceeding the maximum number of stack calls allowed. I also encountered a heap memory exceeded error, which I fixed by increasing the heap memory limit for javascript.
> 
> Sorry for the hiccup with this dependency. We hadn't thought about how to vet small dependencies like this one before, so we're figuring this out on the fly

I compiled the dependency and compared with remote dist using checksum and the hash generated was same.
So i have pinned the workflow to verfied commit :
```- uses: ArcticLampyrid/action-wait-for-workflow@7226d2d248ed9b7e231de31209d4f5f5d4c30295```

Mayank77maruti commented:
@U8NWXD PTAL
Thanks.

Mayank77maruti committed:
Update comment_on_ci_workflow.yml

Mayank77maruti committed:
Update comment_on_ci_workflow.yml

U8NWXD commented on .github/workflows/comment_on_ci_workflow.yml:
```diff
@@ -0,0 +1,68 @@
+          - test_name: "Full-Stack Tests"
+            workflow_file: "full_stack_tests.yml"
+            message: >
+              🛠️ Hi @${{ github.event.pull_request.user.login }}, it looks like the Full-Stack Tests are failing on your PR.
```
```suggestion
              🛠️ Hi @${{ github.event.pull_request.user.login }}, it looks like the full-stack tests are failing on your PR.
```

nit: match the capitalization style used for the rest of the messages

Mayank77maruti committed:
Update .github/workflows/comment_on_ci_workflow.yml

Co-authored-by: U8N WXD <U8NWXD@users.noreply.github.com>

Mayank77maruti commented:
Hey @HardikGoyal2003 this pr has been approved by the assigned reviewer.
Can you please do look into it and merge.

Mayank77maruti commented:
Hey @seanlip this pr has been approved.
Could you please merge it.

seanlip commented:
Just a note, it looks like this is working well (example: https://github.com/oppia/oppia/pull/22129#issuecomment-2722851685).

Congrats @Mayank77maruti -- great job on resolving this! :tada:
